scene/resources/2d/tile_set.cpp

```diff
@@ -6486,21 +6486,38 @@ Ref<NavigationPolygon> TileData::get_navigation_polygon(int p_layer_id, bool p_f
 		Ref<NavigationPolygon> transformed_polygon;
 		transformed_polygon.instantiate();
 
-		PackedVector2Array new_points = get_transformed_vertices(layer_tile_data.navigation_polygon->get_vertices(), p_flip_h, p_flip_v, p_transpose);
+		PackedVector2Array new_points;
+		Vector<PackedInt32Array> new_polygons;
+
+		// Transform polygons
+		for (int i = 0; i < layer_tile_data.navigation_polygon->get_polygon_count(); ++i) {
+			PackedVector2Array new_polygon;
+			PackedInt32Array new_polygon_indices;
+			// Create the polygon
+			auto polygon_indices = layer_tile_data.navigation_polygon->get_polygon(i);
+			for (int j = 0; j < polygon_indices.size(); ++j) {
+				new_polygon.push_back(layer_tile_data.navigation_polygon->get_vertices()[polygon_indices[j]]);
```

Comment:
The vertices should be fetched before as a variable and reused rather than called inside the loop